swift-recon.py

```diff
@@ -351,9 +348,18 @@ def main():
             maas_common.status_err('no ring provided to check')
         stats = swift_replication(args.ring)
     else:
-        maas_common.status_err(
-            'unrecognized command "{0}"'.format(args.recon)
-            )
+        raise TypeError('unrecognized command "{0}"'.format(args.recon))
+    return stats
+
+
+def main():
+    parser = make_parser()
+    args = parser.parse_args()
+
+    try:
+        stats = get_stats_from(args)
+    except (ValueError, TypeError) as e:
```

Comment:
I think wonky custom exceptions are preferable to catching built-in exceptions from large parts of the code.

I think making a base exception in maas_common and using that would be sufficient. I don't think you need to create a load of specific exceptions in this file.